nireports/assembler/reportlet.py

```diff
@@ -270,6 +270,26 @@ def __init__(self, layout, config=None, out_dir=None, bids_filters=None, metadat
                         name=html_anchor,
                         style="; ".join(f"{k}: {v}" for k, v in style.items()),
                     )
+                elif ext in (".png", ".jpg", ".jpeg"):
+                    entities = dict(bidsfile.entities)
+                    if desc_text:
+                        desc_text = desc_text.format(**entities)
+
+                    try:
+                        html_anchor = src.relative_to(out_dir)
+                    except ValueError:
+                        html_anchor = src.relative_to(Path(layout.root))
+                        dst = out_dir / html_anchor
+                        dst.parent.mkdir(parents=True, exist_ok=True)
+                        copyfile(src, dst, copy=True, use_hardlink=True)
+
+                    style = {"width": "100%"} if is_static else {}
+                    style.update(config.get("style", {}))
+
+                    stylestr = "; ".join(f"{k}: {v}" for k, v in style.items())
+                    contents = f'<img src="{html_anchor}" style="{stylestr}"/>'
+                else:
+                    raise RuntimeError(f"Unsupported file extension: {ext}")
```

Comment:
Not sure how folks will feel about this change. I can drop it or change it to a warning.